nyxx.interactions/lib/src/Events/InteractionEvent.dart

```diff
@@ -9,44 +9,44 @@ class InteractionEvent {
   InteractionEvent._new(Nyxx client, Map<String, dynamic> rawJson) {
     this._client = client;
     interaction = Interaction._new(client, rawJson);
+
+    if(interaction.type == 1) {
+      this._pong();
+    }
   }
 
   /// Should be sent when you receive a ping from interactions. Used to acknowledge a ping.
-  Future Pong() {
+  Future _pong() {
     if (DateTime.now()
         .isBefore(this.recievedAt.add(const Duration(minutes: 15)))) {
-      String url;
-      if (hasResponded) {
-        url =
-            "/interactions/${this.interaction.id.toString()}/${this.interaction.token}";
-      } else {
-        url =
-            "/interactions/${this.interaction.id.toString()}/${this.interaction.token}/callback";
-      }
-      final response = this._client.httpEndpoints.sendRawRequest(
-        url,
-        "POST",
-        body: {
-          "type": 1,
-          "data": null,
-        },
-      );
+      if (!hasResponded) {
+        final response = this._client.httpEndpoints.sendRawRequest(
+          "/interactions/${this.interaction.id.toString()}/${this.interaction.token}/callback",
+          "POST",
+          body: {
+            "type": 1,
+            "data": null,
+          },
+        );
 
-      if (response is HttpResponseError) {
-        return Future.error(response);
-      }
+        if (response is HttpResponseError) {
+          return Future.error(response);
+        }
 
-      if (!hasResponded) {
-        hasResponded = true;
+        if (!hasResponded) {
+          hasResponded = true;
+        }
+        return Future.value(null);
```

Comment:
Please do not return null. In that case I think i could be just removed

Reply:
Done